### PR TITLE
nf-core 3.3.0 release

### PR DESCRIPTION
**Pull request by jfy133**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/mag/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/mag _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core pipelines lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Review by sateeshperi — APPROVED** [review]
minor suggestions. 

Looks good overall. I would aim to have to local modules & sub-wfs nf-tests by next release.

**Comment by jfy133** [comment]
> minor suggestions.
> 
> Looks good overall. I would aim to have to local modules & sub-wfs nf-tests by next release.

Slowly slowly getting there 😅 most of the local modules are being ripped and replaced, and I need pipeline tests done first becuase we are getting so many new contributors/ions, but I need to [rejig the tests first](https://github.com/nf-core/mag/pull/621) , but that keeps getting pushed because we then get more bugs/contributors :sweat_smile: (I swear this pipeline is a full time job...)

File: subworkflows/local/utils_nfcore_mag_pipeline/main.nf [hunk 1]
```diff
@@ -26,7 +26,7 @@ workflow PIPELINE_INITIALISATION {
     take:
     version           // boolean: Display version and exit
     validate_params   // boolean: Boolean whether to validate parameters against the schema at runtime
-    monochrome_logs   // boolean: Do not use coloured log outputs
+    _monochrome_logs  // boolean: Do not use coloured log outputs
```
Comment (charles-plessy) [inline subworkflows/local/utils_nfcore_mag_pipeline/main.nf:29]: It took me time to understand that the underscore signals that the variable is not used.  But why not just remove it in that case?

**Review by charles-plessy — COMMENTED** [review]
I looked at all the changes except for the nf-core modules and subworkflows, which I expect to be exact copies from nf-core/modules.

File: docs/output.md [hunk 2]
```diff
@@ -109,25 +109,39 @@ The pipeline uses Nanolyse to map the reads against the Lambda phage and removes
 
 </details>
 
-### Filtlong and porechop
+### Long read adapter removal
 
-The pipeline uses filtlong and porechop to perform quality control of the long reads that are eventually provided with the TSV input file.
+The pipeline uses porecho_abi or porechop to perform adapter trimming of the long reads that are eventually provided with the TSV input file.
```
Comment (jfy133) [inline docs/output.md:114]:
```suggestion
The pipeline uses porechop_abi or porechop to perform adapter trimming of the long reads that are eventually provided with the TSV input file.
```

File: subworkflows/local/utils_nfcore_mag_pipeline/main.nf (hunk 1 above)
Comment (jfy133) [reply subworkflows/local/utils_nfcore_mag_pipeline/main.nf:29]: This is something new from the nextflow formatter (in the nextflow Vscode extension). 

Yes, it's for indicating unused variables, let's say like in a map function where you try to reduce the element of a channel object (e.g. you have meta, bam, Bai, but you only need a bam) but you still need to name each component.

In this particular case however the monochrome logs from from the nf-core tempate so I don't remove it.

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,41 @@
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## 3.3.0 [2024-12-16]
```
Comment (jfy133) [inline CHANGELOG.md:6]:
```suggestion
## 3.3.0 [2024-12-19]
```

**Review by charles-plessy — APPROVED** [review]
Thanks for the changes and explanations.

**Pull request merged**